.github/workflows/validations.yml

```diff
@@ -0,0 +1,69 @@
+name: "Validations"
+on: [push, pull_request]
+jobs:
+  validate-code-style:
+    name: PSR-2 Code style
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Validate composer.json and composer.lock
+        run: composer validate
+
+      - name: Get Composer Cache Directory
+        id: composer-cache
+        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+      - name: Cache composer dependencies
+        uses: actions/cache@v1
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
+          restore-keys: ${{ runner.os }}-composer-
+
+      - name: Install PHP_CodeSniffer composer package
+        run: composer global require "squizlabs/php_codesniffer=*"
+
+      - name: Validate PSR-2 Code Style
+        run: ~/.composer/vendor/bin/phpcs
+
+  validate-code-static:
+    name: Static Code with PHP ${{ matrix.php-versions }}
+    runs-on: ubuntu-latest
+    needs: validate-code-style
+    strategy:
+      fail-fast: false
+      matrix:
+        php-versions: ['7.2', '7.3', '7.4']
```

Comment:
Maybe running the validation the latest is good enough?

Reply:
To make sure that code is valid for every version. Example: deprecation warnings.